src/classes/Keyboard.cpp

```diff
@@ -719,6 +766,30 @@ bool LedKeyboard::setStartupMode(StartupMode startupMode) {
 	return sendDataInternal(data);
 }
 
+bool LedKeyboard::setOnBoardMode(OnBoardMode onBoardMode) {
+	byte_buffer_t data;
+	switch (currentDevice.model) {
+		case KeyboardModel::g815:
+			switch (onBoardMode) {
```

Comment:
This inner switch statement seems unnecessary. You can use the raw uint8_t value of onBoardMode as data[5] (or through a vector push at the end) and eliminate several lines of code here.

Reply:
Do you mean data[4] `0x01` and `0x02` ?  Also note that data[3] is different `0x1a` and `0x1a`.  I'm not sure how to do that without a switch.

Reply:
Yes, that value, since you are using the actual value of the OnBoardMode parameter. You can simply pass this, it actually helps with readability, so you know the field being used.

Have you investigated the value in data[3] to ensure that those exact values matter. I have seen instances where it doesn't. But if it matters, then the need for the conditional remains. Depending on investigation and implementation of other devices, it may even be worth inverting the switch statements in order to reduce redundant code. That could be tackled later if relevant and needed.

Reply:
Alternatively, you could change the data type to uint16_t and do bitshifts to simplify things a little.

Reply:
@pearsonk - yes you're right.  They both work with `11ff111a` followed by `01` or `02`.

I pushed a new commit, unsquished.  Once it's all ok I can squish it.